src/uta/loading.py

```diff
@@ -288,40 +288,80 @@ def load_exonset(session, opts, cf):
 
     update_period = 25
 
-    session.execute(text("set role {admin_role};".format(
-        admin_role=cf.get("uta", "admin_role"))))
+    session.execute(
+        text("set role {admin_role};".format(admin_role=cf.get("uta", "admin_role")))
+    )
     session.execute(text("set search_path = " + usam.schema_name))
 
-    n_rows = len(gzip.open(opts["FILE"], 'rt').readlines()) - 1
-    esr = ufes.ExonSetReader(gzip.open(opts["FILE"], 'rt'))
+    n_rows = len(gzip.open(opts["FILE"], "rt").readlines()) - 1
+    esr = ufes.ExonSetReader(gzip.open(opts["FILE"], "rt"))
     logger.info("opened " + opts["FILE"])
 
     n_new = 0
     n_unchanged = 0
     n_deprecated = 0
+    n_skipped = 0
     n_errors = 0
     for i_es, es in enumerate(esr):
+        skipped = False
         try:
-            n, o = _upsert_exon_set_record(session, es.tx_ac, es.alt_ac, es.strand, es.method, es.exons_se_i)
+            # determine if alignment and transcript have the same exon structure
+            tx_es = (
+                session.query(usam.ExonSet)
+                .filter(
+                    usam.ExonSet.tx_ac == es.tx_ac,
+                    usam.ExonSet.alt_ac == es.tx_ac,
+                    usam.ExonSet.alt_aln_method == "transcript",
+                )
+                .one()
+            )
+            tx_exon_count = len(tx_es.exons_se_i())
+            aln_exon_count = len(es.exons_se_i.split(";"))
+            if tx_exon_count != aln_exon_count:
+                logger.warning(
+                    "Exon structure mismatch: {tx_exon_count} exons in transcript {es.tx_ac}; {aln_exon_count} in alignment {es.alt_ac}".format(
+                        tx_exon_count=tx_exon_count,
+                        aln_exon_count=aln_exon_count,
+                        es=es,
+                    )
+                )
+                skipped = True
+                continue
+
+            n, o = _upsert_exon_set_record(
+                session, es.tx_ac, es.alt_ac, es.strand, es.method, es.exons_se_i
+            )
             session.commit()
         except IntegrityError as e:
             logger.exception(e)
             session.rollback()
             n_errors += 1
+        else:
```

Comment:
I used "else" instead of finally. If there is an exception, then we will not know if skipped or n or o were set properly. We should record the error and move on.